### DISCUSSION TITLE
General Query: Training for text detection for PaddleOCR

### DISCUSSION THREAD

**Discussion by VishyAnand28** [PaddlePaddle/PaddleOCR#13963] [Q&A] [answered]
Hello guys,

Fine-tuning of both text detection and recognition for default algorithms as found here:https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/finetune_en.md is amazingly written and I was able to execute it. DB and SVTR_LCNet are default models for fine-tuning for detection and recognition respectively.

Brief Overview of models employed:
Text Detection: en_PP-OCRv3_det
Text Recognition: en_PP-OCRv4_rec
Corresponding models and yaml file can be found here: https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_ch/models_list.md

I would like to know fine-tune with text detection SAST:

Could I simply use wget -P ./pretrain_models/ https://paddleocr.bj.bcebos.com/pretrained/ResNet50_vd_ssld_pretrained.pdparams as pretrained model from here:https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/detection_en.md with config file from here: https://github.com/PaddlePaddle/PaddleOCR/blob/main/configs/det/det_r50_vd_sast_icdar15.yml? (I want to fine-tune not train from scratch).

Fine-tuning of text recognition with VisionLAN. I have been exploring this for weeks but has left me confused on how to proceed.
2.1 As per https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/recognition_en.md, we can download the pretrained model with wget -P ./pretrain_models/ https://paddleocr.bj.bcebos.com/PP-OCRv4/english/en_PP-OCRv4_rec_train.tar, which is the en_PP-OCRv4_rec model. This step is pretty clear.
2.2 But when I had a look at https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/algorithm_rec_visionlan_en.md, it says that pretrained, trained model can be downloaded from this:
![image](https://github.com/user-attachments/assets/47284695-7ba7-49bc-be30-4de2c15eefa8)

It is also mentioned on VisionLAN that PaddleOCR modularizes the code, and training different recognition models only requires changing the configuration file.

Information on 2.1 and 2,2 is bit confusing to me. Which model should I use to fine-tune on VisionLAN?

Thanks in advance for guidance!

@GreatV @WenmuZhou @LDOUBLEV @MissPenguin @tink2123 @UserWangZz ..... can someone please help me?

**Comment by UserWangZz** [comment]
Can you list your problem, Thks

**Comment by VishyAnand28** [comment]
Thanks for replying, it means a lot @UserWangZz.

How to fine-tune SAST in text detection and VisionLAN in text recognition? 

I have successfully carried out fine-tuning as given on the website with DB and SVTR_LCNet as default models for text detection and recognition respectively but I would like to know for SAST, VisionLAN. I request you to please guide me on this. 

Thanks in advance!

**Reply by UserWangZz** [reply, marked as answer]
You should be able to fine-tune it successfully by following the steps in the documentation, but I don't know exactly what problem you are experiencing, my suggestion is to use the connection mentioned in the documentation to download the pre-trained model for fine-tuning. It's just that using the backbone model may cause the pre-trained parameters to be loaded incorrectly.

**Reply by VishyAnand28** [reply]
Okay, thanks a lot for replying, so in the Config file of text detection, all I have to change is from DB to SAST? 
Could we use the standard config file from here: https://github.com/PaddlePaddle/PaddleOCR/blob/main/configs/det/ch_PP-OCRv3/ch_PP-OCRv3_det_student.yml?

**Reply by VishyAnand28** [reply]
@UserWangZz can you please reply on this? sorry for the disturbance caused, if any!

**Reply by UserWangZz** [reply]
If you need to fine tune the SAST model, you should still use it https://github.com/PaddlePaddle/PaddleOCR/blob/main/configs/det/det_r50_vd_sast_icdar15.yml Configure the file, then find the pre trained model of SAST in the document, and replace the pretrained-model in the configuration file with the address of the pre trained model

**Reply by VishyAnand28** [reply]
@UserWangZz thanks a ton for your time and patience. It means a lot to me, honestly.

Yes, I can find the pre-trained model of the SAST ICDAR2015 dataset on the link: https://github.com/PaddlePaddle/PaddleOCR/blob/main/docs/algorithm/text_detection/algorithm_det_sast.en.md

![image](https://github.com/user-attachments/assets/2c183107-42b2-4939-a9d5-322f905f679f)

So, in your context, when you mean pre-trained model, you mean the trained model present on the above SAST document, right?

**Reply by UserWangZz** [reply]
yes

**Reply by VishyAnand28** [reply]
Thanks a ton once again, @UserWangZz  have a great week ahead.
